### PR TITLE
OpenStack: Add Validate conditions at openstack-provision-etcd-disk-speed step

### PR DESCRIPTION
**Pull request by MiguelCarpio**
Add waiting time to update to false the Progressing state of the etcd cluster.

Add validate the controlPlaneHardwareSpeed value and Etcd Available and Degraded state after the slow profile patch.

### CODE REVIEW

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-ovn-serial

File: ci-operator/step-registry/openstack/provision/etcd-disk-speed/openstack-provision-etcd-disk-speed-commands.sh [hunk 1]
```diff
@@ -9,20 +9,16 @@ function info() {
...
-    if [[ "${ETCD_CO_AVAILABLE}" == "True" ]]; then
-      echo "Patched successfully!"
-      break
-    fi
-    sleep 15
-  done
-  if [[ "${ETCD_CO_AVAILABLE}" != "True" ]]; then
-          echo "Etcd patch failed..."
-          exit 1
+  info 'INFO: Waiting for patch etcd cluster...'
```
Comment (mandre) [inline ci-operator/step-registry/openstack/provision/etcd-disk-speed/openstack-provision-etcd-disk-speed-commands.sh:12]: Nit: Please remove the INFO prefix, it's already in the output of the info() function (on [line 8](https://github.com/openshift/release/pull/59057/files#diff-61a78a69fc41150f08b8b255492a18d34f09d942945bd0bc0f94ae9656d3bec3R8)).

File: ci-operator/step-registry/openstack/provision/etcd-disk-speed/openstack-provision-etcd-disk-speed-commands.sh [hunk 2]
```diff
@@ -9,20 +9,16 @@ function info() {
...
+  if
+    ! oc wait --timeout=1m --for=jsonpath='{.spec.controlPlaneHardwareSpeed}'="Slower" etcd cluster 1>/dev/null || \
+    ! oc wait --timeout=30m --all --for=condition=Progressing=false co etcd 1>/dev/null || \
+    ! oc wait --timeout=1m --all --for=condition=Degraded=false co etcd 1>/dev/null || \
+    ! oc wait --timeout=1m --all --for=condition=Available=true co etcd 1>/dev/null; then
+      info "ERROR: Etcd patch failed..."
+      exit 1
   fi
-  info 'Etcd Patched successfully!'
+  info 'INFO: Etcd Patched successfully!'
```
Comment (mandre) [inline ci-operator/step-registry/openstack/provision/etcd-disk-speed/openstack-provision-etcd-disk-speed-commands.sh:21]: ditto

**Comment by mandre** [comment]
We should try to update the [check in openstack-rotate-cloud-credentials-commands.sh](https://github.com/openshift/release/blob/master/ci-operator/step-registry/openstack/rotate/cloud-credentials/openstack-rotate-cloud-credentials-commands.sh#L61) from:
```
oc wait --timeout=5m --for=condition=Progressing=false $(oc get clusteroperator -o NAME) -o template='{{.metadata.name}} is ready
```

to:
```
oc wait clusteroperators --all --for=condition=Progressing=false --timeout=30m
```

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-ovn-serial

**Review by mandre — CHANGES_REQUESTED** [review]
Let's rework the commit message to match the conventions and limit the title to 50 chars if possible.

https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project for reference.

Also, we should run the CI on the most recent commit before approving.

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.18-techpreview-e2e-openstack-ovn-serial-techpreview

**Review by mandre — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**